cql3/statements/create_function_statement.cc

```diff
@@ -50,7 +50,7 @@ shared_ptr<functions::function> create_function_statement::create(query_processo
        // FIXME: need better way to test wasm compilation without real_database()
        wasm::context ctx{db.real_database().wasm_engine(), _name.name, qp.get_wasm_instance_cache(), db.get_config().wasm_udf_yield_fuel(), db.get_config().wasm_udf_total_fuel()};
        try {
-            wasm::precompile(ctx, arg_names, _body);
+            wasm::precompile(ctx, arg_names, rust::Slice<const rust::u8>(reinterpret_cast<const rust::u8*>(_body.data()), _body.size()));
```

Comment:
Maybe I'm missing the complications here, but why instead of std::string you went all the way to this rust-specific rust::Slice thing? Couldn't you have used std::string_view, and hopefully (?)  that can be trivially converted later to what you want it to be?

Reply:
To compile the script, we have to pass it as an argument to a Rust function, which is why, sooner or later, we have to create a Rust-compatible type. Before the patch it was `rust::Str`, and now it's `rust::Slice` (unfortunately neither can be trivially converted to from std::string_view)

I changed the `Str` to `Slice<u8>`, because the actual compilation method exported by the wasm runtime takes bytes as an argument. The compilation will succeed if the bytes represent a WAT string, or if they are the contents of the .wasm binary. Before, we only had literal WAT strings, so we could construct a `Str` and later convert it to `Slice<u8>` in Rust, but now when we have base64, if we pass it as a string and reinterpret it as bytes, the WASM runtime will not recognize it - the string needs to be decoded first.
 
We could keep the `Str` and do the conversion to bytes in Rust, but for that we would need to pass an additional argument telling which kind of string are we passing, but it seemed like just passing a prepared `Slice` was less complex.